src/connection_check.py

```diff
@@ -53,6 +53,35 @@ def broker_active(unit: Unit, zookeeper_config: Dict[str, str]) -> bool:
         True if broker id is recognised as active by ZooKeeper. Otherwise False.
     """
     broker_id = unit.name.split("/")[1]
+    brokers = brokers_active_get(zookeeper_config=zookeeper_config)
+    chroot = zookeeper_config.get("chroot", "")
+    result = broker_active_check(chroot=chroot, broker_id=broker_id, brokers=brokers)
+    return result
+
+
+def broker_active_check(chroot: str, broker_id: str, brokers: List[str]) -> bool:
+    """Checks ZooKeeper for client connections, checks for specific broker id.
+
+    Args:
+        chroot: chroot config from zookeeper
+        broker_id: the `Unit` to check connection of
+        brokers: the brokers relation data provided by ZooKeeper
+
+    Returns:
+        True if broker id is recognised as active by ZooKeeper. Otherwise False.
+    """
+    return f"{chroot}/brokers/ids/{broker_id}" in brokers
+
+
+def brokers_active_get(zookeeper_config: Dict[str, str]) -> List[str]:
```

Comment:
This is a good function to have, great addition!

Reply:
Thanks!